src/Instrumentation/PDO/src/PDOAttributeTracker.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Contrib\Instrumentation\PDO;
+
+use OpenTelemetry\SemConv\TraceAttributes;
+
+final class PDOAttributeTracker
+{
+    private $pdoToAttributesMap;
+    private $statementMapToPdoMap;
+
+    public function __construct()
+    {
+        $this->pdoToAttributesMap = [];
+        $this->statementMapToPdoMap = [];
+    }
+
+    public function trackStatementToPdoMapping(\PDOStatement $statement, \PDO $pdo)
+    {
+        $this->statementMapToPdoMap[spl_object_id($statement)] = spl_object_id($pdo);
+    }
+
+    public function removeMapping(\PDOStatement $statement)
+    {
+        $key = spl_object_id($statement);
+        if (array_key_exists($key, $this->statementMapToPdoMap)) {
+            unset($this->statementMapToPdoMap[$key]);
+        }
+    }
+
+    /**
+     * Maps a statement back to the connection attributes.
+     *
+     * @param \PDOStatement $statement
+     * @return iterable<non-empty-string, bool|int|float|string|array|null>
+     */
+    public function trackedAttributesForStatement(\PDOStatement $statement): iterable
+    {
+        $statementKey = spl_object_id($statement);
+        if (!array_key_exists($statementKey, $this->statementMapToPdoMap)) {
+            return [];
+        }
+
+        $pdoKey = $this->statementMapToPdoMap[$statementKey];
+        if (!array_key_exists($pdoKey, $this->pdoToAttributesMap)) {
+            return [];
+        }
+
+        return $this->pdoToAttributesMap[$pdoKey];
```

Comment:
```suggestion
        return $this->pdoToAttributesMap[$this->statementMapToPdoMap[spl_object_id($statement)] ?? null] ?? [];
```

Reply:
i dont feel like the suggested change is very readable